prow/github/hmac.go

```diff
@@ -46,3 +64,37 @@ func PayloadSignature(payload []byte, key []byte) string {
 	sum := mac.Sum(nil)
 	return "sha1=" + hex.EncodeToString(sum)
 }
+
+// For given repository/organization, returns all *valid* HMAC tokens. Validity is determined based on time till they are valid.
+func ExtractHmacs(repo string, tokenGenerator func() []byte) [][]byte {
+	t := tokenGenerator()
+	repoToTokenMap := map[string]map[string]time.Time{}
+
+	err := yaml.Unmarshal(t, &repoToTokenMap)
+	if err != nil {
+		fmt.Printf("error: %v", err)
+		return nil
+	}
+
+	orgName := strings.Split(repo, "/")[0]
+	var extractValidToken = func(allTokens map[string]time.Time) [][]byte {
```

Comment:
This returns multiple tokens, so `extractValidTokens`.

Also, why is this an inner function instead of just a private top-level one?

Reply:
Renamed. I wanted to keep this an inner function as it has limited scope and its a short one. Don't have a strong inclination one way or the other.